sklearn/utils/_tags.py

```diff
@@ -1,71 +1,318 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+
+import numpy as np
+
 # Authors: The scikit-learn developers
 # SPDX-License-Identifier: BSD-3-Clause
 
-import numpy as np
 
-_DEFAULT_TAGS = {
-    "array_api_support": False,
-    "non_deterministic": False,
-    "requires_positive_X": False,
-    "requires_positive_y": False,
-    "X_types": ["2darray"],
-    "poor_score": False,
-    "no_validation": False,
-    "multioutput": False,
-    "allow_nan": False,
-    "stateless": False,
-    "multilabel": False,
-    "_skip_test": False,
-    "_xfail_checks": False,
-    "multioutput_only": False,
-    "binary_only": False,
-    "requires_fit": True,
-    "preserves_dtype": [np.float64],
-    "requires_y": False,
-    "pairwise": False,
-}
-
-
-def _safe_tags(estimator, key=None):
-    """Safely get estimator tags.
+@dataclass
+class InputTags:
```

Comment:
I think these need to be added to `doc/api_reference.py` to make them show up in the docs.